processor/src/main/java/org/mapstruct/ap/internal/util/Filters.java

```diff
@@ -111,7 +114,24 @@ public Map<String, ReadAccessor> recordAccessorsIn(Collection<Element> recordCom
     }
 
     private TypeMirror getReturnType(ExecutableElement executableElement) {
-        return getWithinContext( executableElement ).getReturnType();
+        TypeMirror returnType = getWithinContext( executableElement ).getReturnType();
```

Comment:
rename `getWithinContext` to `getMethodAsMemberOfDefiningClass`. guess thats a more appropriate name